maxprocs/maxprocs.go

```diff
@@ -103,17 +114,92 @@ func SetWithConfig(c *config.Config, opts ...Option) {
 		minProcs:              c.GetInt("MinProcs", defaultMinProcs),
 		cpuRequestsMultiplier: c.GetFloat64("RequestsMultiplier", defaultCPURequestsMultiplier),
```

Comment:
I believe we are missing a guard to not exceed the total number of available cpus in the node when setting `GOMAXPROCS` after doing `cpuRequests*cpuRequestsMultiplier`

Reply:
I'll add a `maxProcs` same a `minProcs`:

```go
type conf struct {
	logger                logger.Logger
	minProcs              int
	maxProcs              int // NEW ATTRIBUTE
	cpuRequestsMultiplier float64
	roundQuotaFunc        func(float64) int
	stop                  chan os.Signal
}
```

Reply:
Do we need this to be configurable? I was just thinking `runtime.NumCPU()`